### PR TITLE
Set log level of individual loggers

### PR DESCRIPTION
**Pull request by rvl**
# Overview

Currently if debug logging is enabled (`--verbose` option), there is way too much information logged. This modifies the CLI options to allow setting the log levels of individual loggers.

- Modifies verbose/quiet CLI options to accept a logger name.
- Updates iohk-monitoring framework version
- Begin adopting some logging patterns such as DefineSeverity.

# Comments

- There are better ways to enable/disable loggers (e.g. how it's done in cardano-node), but that will be more work to implement. So this is just the fastest way of doing it.

### CODE REVIEW

File: lib/cli/src/Cardano/CLI.hs [hunk 1]
```diff
@@ -1217,57 +1228,65 @@ instance FromText TxId where
...
+
+allTracers :: [Text]
+allTracers = map ("cardano-wallet" <>)
+    [ ".api"
+    , ".daedalus-ipc"
+    , ".database"
+    , ".database.query"
+    , ".stake-pools"
+    , ".worker"
+    , "" ]
```
Comment (KtorZ) [inline lib/cli/src/Cardano/CLI.hs:1251]: I suppose this last one works like a wildcard? I haven't played with it but I wonder how we can pass in an empty tracer like this? `--quiet` or `--verbose` "naked" (with no argument) ?

File: lib/cli/src/Cardano/CLI.hs [hunk 2]
```diff
@@ -1217,57 +1228,65 @@ instance FromText TxId where
...
 -- | Map a given 'Verbosity' level onto a 'Severity' level.
 verbosityToMinSeverity :: Verbosity -> Severity
 verbosityToMinSeverity = \case
-    Default -> Info
-    Quiet   -> Error
     Verbose -> Debug
+    Quiet -> Error
+
+allTracers :: [Text]
+allTracers = map ("cardano-wallet" <>)
```
Comment (KtorZ) [inline lib/cli/src/Cardano/CLI.hs:1244]: Wouldn't it be better from a user experience that the "cardano-wallet" is added on top of the user inputs? As far as I understand it, if I want to make the api more verbose, I'll do: `cardano-wallet ... --verbose cardano-wallet.api` whereas, it sounds slightly better to only have to do `cardano-wallet ... --verbose api`, the `cardano-wallet` is already implied by the surrounding context. No ?

**Comment by rvl** [comment]
> I suppose this last one works like a wildcard? I haven't played with it but I wonder how we can pass in an empty tracer like this? --quiet or --verbose "naked" (with no argument) ?
> Wouldn't it be better from a user experience that the "cardano-wallet" is added on top of the user inputs? As far as I understand it, if I want to make the api more verbose, I'll do: cardano-wallet ... --verbose cardano-wallet.api whereas, it sounds slightly better to only have to do cardano-wallet ... --verbose api, the cardano-wallet is already implied by the surrounding context. No ?

There has to be an argument on these CLI switches for parsing reasons. The wildcard could be something like `*` or `all`. I agree that removing the `cardano-wallet.` prefix is better for the user. However we have a log of logging using that top-level tracer name.

What cardano-node have done is defined a few components which they can enable/disable logging for. The cli switches are something like `--enable-xyz-logger`. There is no configurable log level that I can see, except through `logging-config.yaml`. And "verbose" has a different meaning there. It corresponds to the amount of detail/data in the log messages.

To get a short term solution, we could define more trace names and log far fewer traces under the top-level `cardano-wallet` tracer name.

**Comment by rvl** [comment]
I might put this on ice until we have a little time to do the `Tracers` record data type as used in [ouroboros-consensus](https://github.com/input-output-hk/ouroboros-network/blob/5941c192b64f39d1ab276e0a747a0e47fb52cef8/ouroboros-consensus/src/Ouroboros/Consensus/Node/Tracers.hs#L37).

**Comment by KtorZ** [comment]
> To get a short term solution, we could define more trace names and log far fewer traces under the top-level cardano-wallet tracer name.

Indeed. Starting a fresh new wallet though, I observes "only" these log lines under the top-level `cardano-wallet`

**launch**
```
[iohk.cardano-wallet:Info:ThreadId 4] [2019-11-25 08:16:49.19 UTC] Using directory: /home/ktorz/.local/share/cardano-wallet/jormungandr/testnet
[iohk.cardano-wallet:Info:ThreadId 4] [2019-11-25 08:16:49.19 UTC] Using directory: /home/ktorz/.local/share/cardano-wallet/jormungandr/testnet/wallets
[iohk.cardano-wallet:Info:ThreadId 4] [2019-11-25 08:16:49.19 UTC] Running as v2019.11.18 (git revision: bc8cebf2a266fed0733812ef981c9825975fe0aa)
[iohk.cardano-wallet:Info:ThreadId 4] [2019-11-25 08:16:49.19 UTC] Wallet backend server starting...
[iohk.cardano-wallet:Info:ThreadId 4] [2019-11-25 08:16:49.19 UTC] Node is Jörmungandr on testnet
[iohk.cardano-wallet:Info:ThreadId 32] [2019-11-25 08:16:50.21 UTC] Waiting for Jörmungandr to be ready on tcp/35857
[iohk.cardano-wallet:Info:ThreadId 32] [2019-11-25 08:16:50.22 UTC] Jörmungandr is ready.
[iohk.cardano-wallet:Info:ThreadId 32] [2019-11-25 08:16:50.23 UTC] Wallet backend server listening on 127.0.0.1:8090
``` 

**serve**
```
[iohk.cardano-wallet:Info:ThreadId 4] [2019-11-25 08:23:09.67 UTC] Running as v2019.11.18 (git revision: bc8cebf2a266fed0733812ef981c9825975fe0aa)
[iohk.cardano-wallet:Info:ThreadId 4] [2019-11-25 08:23:09.67 UTC] Wallet backend server starting...
[iohk.cardano-wallet:Info:ThreadId 4] [2019-11-25 08:23:09.67 UTC] Node is Jörmungandr on testnet
[iohk.cardano-wallet:Info:ThreadId 26] [2019-11-25 08:23:09.67 UTC] Waiting for Jörmungandr to be ready on tcp/45467
[iohk.cardano-wallet:Info:ThreadId 26] [2019-11-25 08:23:09.67 UTC] Jörmungandr is ready
```

Which really occurs during the start-up phase. Turning `--verbose` only increases the API verbosity and, more logs from the database calls (under `iohk.cardano-wallet.database`). I'll add some namespace for the first start-up logs and remove the prefix.

**Comment by KtorZ** [comment]
> I might put this on ice until we have a little time to do the Tracers record data type as used in ouroboros-consensus.

@rvl Okay. I'll open a ticket in JIRA to keep track of this as a story.

**Pull request closed without merging**